solidity/package.json

```diff
@@ -11,57 +11,68 @@
   },
   "devDependencies": {
     "@layerzerolabs/solidity-examples": "^1.1.0",
-    "@nomiclabs/hardhat-ethers": "^2.2.1",
+    "@nomiclabs/hardhat-ethers": "^2.2.3",
     "@nomiclabs/hardhat-waffle": "^2.0.6",
-    "@typechain/ethers-v5": "^10.0.0",
-    "@typechain/hardhat": "^6.0.0",
+    "@typechain/ethers-v5": "^11.1.2",
+    "@typechain/hardhat": "^9.1.0",
     "chai": "^4.3.6",
     "ethereum-waffle": "^4.0.10",
     "ethers": "^5.7.2",
-    "hardhat": "^2.19.0",
+    "hardhat": "^2.22.2",
     "hardhat-gas-reporter": "^1.0.9",
     "prettier": "^2.8.8",
     "prettier-plugin-solidity": "^1.1.3",
     "solhint": "^4.0.0",
     "solhint-plugin-prettier": "^0.0.5",
     "solidity-coverage": "^0.8.3",
     "ts-generator": "^0.1.1",
-    "typechain": "^8.3.2",
-    "typescript": "5.1.6"
+    "ts-node": "^10.8.0",
+    "typechain": "patch:typechain@npm%3A8.3.2#~/.yarn/patches/typechain-npm-8.3.2-b02e27439e.patch",
+    "typescript": "5.3.3"
   },
   "directories": {
     "test": "test"
   },
+  "type": "module",
+  "exports": {
```

Comment:
You can think of exports as the replacement for `main` but also greater flexibility for multiple paths under a single package